### PR TITLE
fix(@aws-amplify-react/auth): handle required attributes in NEW_PASSWORD_REQUIRED challenge

### PR DESCRIPTION
**Pull request by PerfectPixel**
*Issue #, if available:*
Input fields for required attributes configured via CognitoUserPool are missing. https://github.com/aws-amplify/amplify-js/issues/935

*Description of changes:*
Read requiredAttributes array and display the corresponding form fields.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/RequireNewPassword.jsx [hunk 1]
```diff
@@ -59,8 +59,8 @@ export default class RequireNewPassword extends AuthPiece {
     change() {
         const user = this.props.authData;
         const { password } = this.inputs;
-        const { requiredAttributes } = user.challengeParam;
-    
+        const requiredAttributes = objectWithoutProperties(this.inputs, ['password'])
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/RequireNewPassword.jsx]: Can we change this method to `objectWithProperties` for example:
```
const { requiredAttributes } = user.challengeParam;
const attrs = objectWithProperties(this.inputs, requiredAttributes);
```

**Comment by PerfectPixel** [comment]
Sure, technically there will not be any other inputs there, but we can do a whitelist instead of a blacklist. I'll provide an update tomorrow 👍🏼

**Comment by PerfectPixel** [comment]
@powerful23 all done

**Comment by powerful23** [comment]
fixes #935 
fixes #1306

**Pull request merged**